cmd/minishift/cmd/stop.go

```diff
@@ -61,12 +61,16 @@ func runStop(cmd *cobra.Command, args []string) {
 
 	fmt.Println("Stopping local OpenShift cluster...")
 
-	if !skipUnRegistration {
-		if _, err := cluster.UnRegister(api); err != nil {
+	if !skipUnRegistration && minishiftConfig.InstanceConfig.IsRegistered {
+		supportUnregistration, err := cluster.UnRegister(api)
```

Comment:
> If minishiftConfig.InstanceConfig.IsRegistered was the whole point, it seems we neglected to actually check the state of this variable.

That's correct. It should have been checked. Similar to the delete _command_.

> This is why this change is suggested/issue got filed.

Sure. I am not saying there is no issue here. I am just saying that if _minishiftConfig.InstanceConfig.IsRegistered_ is _true_, we already know that _supportUnregistration_ is true.